internal/pkg/varfile/varfile_test.go

```diff
@@ -0,0 +1,215 @@
+package varfile
+
+import (
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/nomad-pack/sdk/pack/variables"
+	"github.com/shoenig/test/must"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func TestVarfile_DecodeHCL(t *testing.T) {
+	type exp struct {
+		dLen  int
+		diags hcl.Diagnostics
+		oLen  int
+		oMap  variables.Overrides
+	}
+	testCases := []struct {
+		name string
+		src  []byte
+		exp  exp
+	}{
+		{
+			name: "empty",
+			src:  []byte{},
+			exp:  exp{},
+		},
+		{
+			name: "comment only",
+			src:  []byte("# just a comment"),
+			exp:  exp{},
+		},
+		{
+			name: "single override",
+			src:  []byte(`mypack.foo = "bar"`),
+			exp: exp{
+				dLen: 0,
+				oLen: 1,
+				oMap: variables.Overrides{
+					"embedded.hcl": []*variables.Override{
+						{
+							Name:  "foo",
+							Path:  "mypack",
+							Type:  cty.String,
+							Value: cty.StringVal("bar"),
+							Range: hcl.Range{
+								Filename: "embedded.hcl",
+								Start:    hcl.Pos{Line: 1, Column: 1, Byte: 0},
+								End:      hcl.Pos{Line: 1, Column: 19, Byte: 18},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "missing equal",
+			src:  []byte(`mypack.foo "bar"`),
+			exp: exp{
+				dLen: 1,
+				diags: hcl.Diagnostics{
+					&hcl.Diagnostic{
+						Severity: hcl.DiagError,
+						Summary:  `Missing key/value separator`,
+						Detail:   `Expected an equals sign ("=") to mark the beginning of the attribute value.`,
+						Subject: &hcl.Range{
+							Filename: "embedded.hcl",
+							Start:    hcl.Pos{Line: 1, Column: 12, Byte: 11},
+							End:      hcl.Pos{Line: 1, Column: 13, Byte: 12},
+						},
+						Context: &hcl.Range{
+							Filename: "embedded.hcl",
+							Start:    hcl.Pos{Line: 1, Column: 1, Byte: 0},
+							End:      hcl.Pos{Line: 1, Column: 13, Byte: 12},
+						},
+						EvalContext: nil,
+						Extra:       DiagExtraFixup{Fixed: true},
+					},
+				},
+			},
+		},
+		{
+			name: "error only",
+			src:  []byte("boom"),
+			exp: exp{
+				dLen: 1,
+				oLen: 0,
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
+			om := make(variables.Overrides)
+			_, diags := Decode("embedded.hcl", tc.src, nil, &om)
+			must.Len(t, tc.exp.dLen, diags, must.Sprintf("slice values: %v", diags))
+
+			if len(tc.exp.diags) > 0 {
+				for i, o := range diags {
+					e := tc.exp.diags[i]
+					must.Eq(t, e, o)
+				}
+			}
+			// There should always be a single element map for these tests having the
+			// filename as the key
+			switch tc.exp.oLen {
+			case 0:
+				must.MapLen[variables.Overrides](t, 0, om, must.Sprintf("map contents: %v", spew.Sdump(om)))
```

Comment:
I think `spew`ing the contents here is maybe a bit much, we could just have a small helper function that would output map contents. I think it's just best to avoid pulling spew as a dependency because it's a big one, but then again I think templates pull it as a dependency anyway, so I don't think strongly about this.